xarray/backends/api.py

```diff
@@ -247,7 +247,7 @@ def maybe_decode_store(store, lock=False):
             token = tokenize(file_arg, group, decode_cf, mask_and_scale,
                              decode_times, concat_characters, decode_coords,
                              engine, chunks, drop_variables)
-            name_prefix = '%s:%s/' % (filename_or_obj, group or '')
+            name_prefix = 'open_dataset-%s:%s/' % (filename_or_obj, group or '')
```

Comment:
Dask splits by hyphen and then keeps all parts that are strictly letters

```python
In [1]: from dask.utils import key_split

In [2]: key_split('foo-bar-123')
Out[2]: 'foo-bar'

In [3]: key_split('foo-bar-/abc123')
Out[3]: 'foo-bar'
```

At a certain point we stopped trying to provide keynames that were indicative of the arguments like `load-/path/to/file` and started just doing `'load-' + tokenize(...)` everywhere.